### PR TITLE
Isaac bullinger gitpod exploration

### PR DESCRIPTION
**Pull request by isaacBullinger**
Made separate pages with update and delete for the database. Added email, registered licenses and city into the add page.  I don't know if it manipulates the data because the backend is disabled. Review thoroughly!

### CODE REVIEW

**Review by andey-robins — CHANGES_REQUESTED** [review]
@isaacBullinger can you make the changes to fix that database back to the right one?

i’ll finish reviewing everything else then! :)

**Comment by isaacBullinger** [comment]
Ok, I made the change. Sorry about that.

**Comment by andey-robins** [comment]
Hey you're good! No need to apologize :) That's why we review it, to make sure that everyone is on the same page!

Next thing to do, let's squash some of these commits. Throwing 25 new commits into the repo will make it difficult for people to understand what changes are happening where and in what places. Try to pick the top two or three commits from this list, give them descriptive names, and then squash the rest into those. [Here's a nice explanation of how (when and why) to squash commits!](https://www.git-tower.com/learn/git/faq/git-squash/).

If you have questions, or just want someone to help as you go through the process since making changes to git is definitely intimidating, ask in the git related slack channel!

@isaacBullinger

**Comment by isaacBullinger** [comment]
I can enter the rebase but whenever I try to squash it doesn't save the file and merge them.

**Comment by andey-robins** [comment]
Let's see if we can summon someone who has more familiarity with gitpod than I do @isaacBullinger 

@dr-mikeb @masonj188 @dradosevich @mthoney

**Comment by isaacBullinger** [comment]
Merge conflict fixed, squash commits on merge.

**Comment by swolf4070** [comment]
^^ There is an option while merging to squash the commits (drop down button next to merge button --> select merge and commit), super easy, no rebase necessary.

File: src/SimpleServer.py [hunk 1]
```diff
@@ -18,28 +18,96 @@ def basic():
 
 @app.route("/Employee/AddEmployee", methods=['POST', 'GET'])
 def studentAddDetails():
+	if request.method =='GET':
+		return render_template('EmployeeData.html')
+	if request.method =='POST':
```
Comment (andey-robins) [inline src/SimpleServer.py:23]: This post method never runs on my end. When I actually click to submit an additional Employee's data, all that the server sees is a GET request with the query string you constructed.

File: src/SimpleServer.py [hunk 2]
```diff
@@ -18,28 +18,96 @@ def basic():
...
+		lastName = request.form.get('lastName', default="Error")
+		businessunit = request.form.get('bu', default="Error")
+		state = request.form.get('state', default="Error")
+		city = request.form.get('city', default="Error") # Added city request.
+		rl = request.form.get('rl', default="Error") # Added registered license request.
+		print("inserting employee"+firstName)
+		try:
+			conn = sqlite3.connect(DATABASE)
+			cur = conn.cursor()
+			cur.execute("INSERT INTO EmployeeList ('FirstName', 'LastName', 'Business Unit', 'State/Province', 'City', 'Registered Licenses')\
```
Comment (andey-robins) [inline src/SimpleServer.py:35]: This SQL string looks right to me, so good there :)

It should just need the above mentioned fix to get the query happening properly

File: src/templates/EmployeeData.html [hunk 3]
```diff
@@ -5,22 +5,28 @@
...
   function loadName() {
+    var email = document.forms["myForm"]["email"].value;
     var firstName = document.forms["myForm"]["firstName"].value;
     var lastName = document.forms["myForm"]["lastName"].value;
     var businessunit = document.forms["myForm"]["bu"].value;
     var state = document.forms["myForm"]["state"].value;
-    params = 'firstName='+firstName+'&lastName='+lastName+'&bu='+businessunit+'&state='+state;
+    var city = document.forms["myForm"]["city"].value; // Edited here
+    var rl = document.forms["myForm"]["rl"].value; // And here
+    params = 'email'+email'&firstName='+firstName+'&lastName='+lastName+'&bu='+businessunit+'&state='+state'&city'+city'&rl'+rl; // Added email, city and registered licenses.
```
Comment (andey-robins) [inline src/templates/EmployeeData.html]: I'm not sure what happened to break the POST options in this file, but perhaps looking at this query string might be a place to start? That's very much a shot in the dark though, cause it looks right to me at first glance.

**Review by andey-robins — CHANGES_REQUESTED** [review]

File: src/SimpleServer.py (hunk 1 above)
Comment (andey-robins) [reply src/SimpleServer.py:23]: image

Comment (andey-robins) [reply src/SimpleServer.py:23]: Above is my server's console when I hit enter on the AddEmployee Page. I believe that immediately following that third entry there needs to be a post request like in the image below (which is what is currently on `main` and supports the insertion of new employees)

image

File: src/templates/EmployeeData.html (hunk 3 above)
Comment (isaacBullinger) [reply src/templates/EmployeeData.html]: I forgot to add some + signs between some of the variables I added in params, it is posting now.

**Review by andey-robins — CHANGES_REQUESTED** [review]
Way to go getting the POST to work!

It still looks like the changes aren't happening in the database. I'll take a look at it during our lunch to debug why that might be happening :)

**Review by andey-robins — APPROVED** [review]
Actually, I just went through to read the issues and saw that this is just a frontend fix :) Good work there @isaacBullinger

**Review by mwodahl — APPROVED** [review]
Looks good

**Pull request merged**